src/nuvolakit-runner/components/lyrics/GecimiLyricsFetcher.vala

```diff
@@ -0,0 +1,116 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+ namespace Nuvola {
+
+
+    /**
+     * Lyric fetcher for [[http://doc.gecimi.com/en/latest/|Gecimi (Lyrics Fans) Lyrics]].
+     * Mainly Chinese and English lyrics.
+     */
+    public class GecimiLyricsFetcher : GLib.Object, LyricsFetcher {
+        public Soup.Session session {get; construct set;}
+        /**
+         * URL format of AZ Lyrics
+         */
+
+        private const string SONG_API = "https://gecimi.com/api/lyric/%s/%s";
+    
+        public GecimiLyricsFetcher(Soup.Session session) {
+            GLib.Object(session: session);
+        }
+    
+        /**
+         * {@inheritDoc}
+         */
+        public async string fetch_lyrics(string artist, string song) throws LyricsError {
+            /* Fetch lrc list */
+            Soup.Message message;
+            string url = SONG_API.printf(transform_name(song), transform_name(artist));
+            message = new Soup.Message("GET", url);
+    
+            string response = (string) message.response_body.flatten().data;
+            if (message.status_code != 200 || response == "") {
+                throw new LyricsError.NOT_FOUND(@"Song $song was not found on Gecimi Lyrics");
+            }
+            
+            /* Get lrc content */
+            var lrc_url = get_lrc_url(response);
+            if (lrc_url == "") {
+                throw new LyricsError.NOT_FOUND(@"Song $song was not found on Gecimi Lyrics");
+            }
+            message = new Soup.Message("GET", lrc_url);
+
+            SourceFunc callback = fetch_lyrics.callback;
+            session.queue_message(message, () => {
+                Idle.add((owned) callback);
+            });
+            yield;
+
+            response = (string) message.response_body.flatten().data;
+            if (message.status_code != 200 || response == "") {
+                throw new LyricsError.NOT_FOUND(@"Song $song was not found on Gecimi Lyrics");
+            }
+            
+            return response;
+        }
+    
+        private string get_lrc_url(string response) {
+            
+            var parser = new Json.Parser();
+            parser.load_from_data (response, -1);
+
+            var root_object = parser.get_root ().get_object();
+            var results = root_object.get_array_member("result");
+            var best_matched = results.get_elements().first().data.get_object();
+            string lrc_url = best_matched.get_string_member("lrc");
```

Comment:
You should add some error checking. `parser.load_from_data` may raise an error. The other code will produce warnings if the JSON structure is different. It should throw errors.